src/amo/pages/AddonReviewList/index.js

```diff
@@ -333,29 +333,36 @@ export class AddonReviewListBase extends React.Component<InternalProps> {
           {reviewId && (
             <FeaturedAddonReview addon={addon} reviewId={reviewId} />
           )}
-          <CardList
-            className="AddonReviewList-reviews-listing"
-            footer={paginator}
-            header={reviewCountHTML}
-          >
-            <ul>
-              {allReviews.map((review, index) => {
-                // TODO: Remove this and use the API to filter out the featured review once
-                // https://github.com/mozilla/addons-server/issues/9424 is fixed.
-                if (
-                  !reviewId ||
-                  (review && review.id.toString() !== reviewId)
-                ) {
-                  return (
-                    <li key={String(index)}>
-                      <AddonReviewCard addon={addon} review={review} />
-                    </li>
-                  );
-                }
-                return null;
-              })}
-            </ul>
-          </CardList>
+          {// TODO: remove this condition after
+          //  https://github.com/mozilla/addons-server/issues/9424 is fixed
+          //  and integrated.
+          (!reviewId ||
```

Comment:
This if statement is pretty complicated. How about just replacing `allReviews.map((review, index) => ...)` (below) with a variable? Then you can simply check the length.

Reply:
I'm not sure I follow what you mean, but I think I can take your idea *and* @rebmullin's idea and combine them. If I filter the array first then I can just check its length. Maybe that's what you meant.

Reply:
Yes, I think that's the same idea.